modules/terraform-aci-contract/main.tf

```diff
@@ -61,3 +62,74 @@ resource "aci_rest_managed" "vzRsSubjGraphAtt" {
     tnVnsAbsGraphName = each.value.service_graph
   }
 }
+
+resource "aci_rest_managed" "vzSubj_unidir" {
+  depends_on = [aci_rest_managed.vzSubj]
+  for_each   = { for subj in var.subjects : subj.name => subj if subj.apply_both_directions == false }
+  dn         = "${aci_rest_managed.vzBrCP.dn}/subj-${each.value.name}"
+  class_name = "vzSubj"
+  content = {
+    name        = each.value.name
+    nameAlias   = each.value.alias
+    descr       = each.value.description
+    revFltPorts = "no"
+    prio        = each.value.qos_class
+    targetDscp  = each.value.target_dscp
+  }
+}
+
+resource "aci_rest_managed" "vzInTerm" {
+  depends_on = [aci_rest_managed.vzSubj_unidir, aci_rest_managed.vzSubj]
+  for_each   = { for subj in var.subjects : subj.name => subj if subj.apply_both_directions == false }
+  dn         = "${aci_rest_managed.vzBrCP.dn}/subj-${each.value.name}/intmnl"
```

Comment:
Dependency is not required here. Instead, define the `dn` as `${aci_rest_managed.vzSubj[each.key].dn}/intmnl`